tests/testthat/test-run_stress_test.R

```diff
@@ -0,0 +1,23 @@
+test_that("is sensitive to `asset_type`, and defaults to bonds", {
+  local_dev_env()
+  expect_equal(run_stress_test("equity"), "equity")
+})
+
+test_that("with invalid `asset_type` errors gracefully", {
+  expect_error(run_stress_test("invalid"), "should be.*bonds.*equity.*loans")
+})
+
+test_that("if `asset_type` is longer than 1 uses default with a warning", {
+  local_dev_env()
+  expect_snapshot(out <- run_stress_test())
+  expect_equal(out, "bonds")
+})
+
+test_that("returns the first argument invisibly", {
+  skip_on_ci()
+  skip_on_cran()
+
+  readRenviron(here::here(".Renviron"))
+  expect_invisible(out <- purrr::quietly(run_stress_test)("bonds"))
```

Comment:
I am not entirely comfortable with having unit tests trigger run_stress_test. While of course it is beneficial to have it as an integration test:

- it would force us to set ST_PROJECT_FOLDER to a bonds compatible project every time we run devtools::test/check which would create overhead
- we had agreed in the team, that outputs in the ST_TESTING projects are always referring to results created from master. If running devtools::check on a dev branch we will mess with that. Strictly speaking this is not a problem, since ST_TESTING is supposed to be an experimental dev folder, but we would have to communicate this change in convention to the team.
- As a compromise: How about setting it in a way, that this test is skipped by default and needs to be explicitly activated?

Reply:
1. That's what my project-specific .Renviron does. Note it points to a directory that nobody else has, so there is no way to mess production directories.

``` r
writeLines(readLines(".Renviron"))
# Where to save project agnostic files?
# For example, see 2DII's Dropbox at
# PortCheck/00_Data/07_AnalysisInputs/ST_INPUTS_MASTER
ST_DATA_PATH=/home/mauro/tmp/ST_INPUTS_MASTER

# Where to save project specific inputs and outputs?
# For example, for bonds (fastest) see 2DII's Dropbox at 
# PortCheck_v2/10_Projects/ST_TESTING_BONDS
ST_PROJECT_FOLDER=/home/mauro/tmp/ST_TESTING_BONDS
```

2. I don't understand this. But it seems very different to what the usual development workflow is. It seems people are using the package in developer-mode to do analyses they should be doing in user-mode. Because the package is now usable in user-mode, analysts should be able to create a project, install this package, call `library()` and start using it. They should not need the source code, and their project should have the .Renviron file pointing wherever they want the IO to live.

I think we should talk in a meeting about this. Because a small change in the workflows might bring a great increase in quality-of-work-life.

3. How about setting it in a way, that this test is skipped by default and needs to be explicitly activated?

Good idea! While I wait to better understand what's the source of the workflow problem, this test will run only in my computer :)

```r
  skip_if_not(is_mauro())
```